enhancements/console/customize-perspectives.md

```diff
@@ -0,0 +1,345 @@
+
+### Goals
+
+- Allow cluster admins to hide the user perspectives.
+- Provide a way that cluster admins can do this based on resource permissions (RBAC).
+
+### Non-Goals
+
+- Validating the console resource in the operator and inform the cluster admins with an error because the operator
+doesn't know the available perspectives provided by the console plugins.
+- Allow cluster admins to customize the navigation menu along with perspectives for users or groups.
+
+## Proposal
+
+We can extend the existing `operator.openshift.io/v1` / `Console` CRD. 
+We can add 
+`spec.customization.perspectives` which can have list of customization details.
+
+### Workflow Description
+
+To disable perspective for users with limited permissions we need a list of options with a perspective id and the required permissions.
+
+```text
+├── Spec
+│   ├── customization
+|       ├── perspectives 
+|           ├── id
+|           ├── visibility
+|               ├── state
+|               ├── accessReview
+|                   ├── required
+|                       ├── group
+|                       ├── resource
+|                       ├── verb
+└── ...
+```
+
+This might be extendable in the future if we want replace (hide one, show another) a perspective based on a resource permissions:
+
+```text
+├── Spec
+│   ├── customization
+|       ├── perspectives 
+|           ├── id
+|           ├── visibility
+|               ├── state
+|               ├── accessReview
+|                   ├── required
+|                   |   ├── group
+|                   |   ├── resource
+|                   |   ├── verb
+|                   ├── missing
+|                       ├── group
+|                       ├── resource
+|                       ├── verb
+└── ...
+```
+
+Each perspective conforms to the following schema:
+
+- `id: string` , defines the id of perspective to disable. Incorrect or unknown ids will be ignored.
+
+- `visibility: object` , defines the state of perspective along with access review checks if needed for that perspective.
+
+- `state: Enabled/Disabled/AccessReview` , defines the perspective is enabled or disabled or access review check is required.
+
+// +optional
+- `accessReview: object`, defines required and missing access review checks.
+
+// +optional
+- `required: authV1.ResourceAttributes[]` , defines a list of permission checks. The perspective will only be shown when all checks are successful.
+
+// +optional
+- `missing: authV1.ResourceAttributes[]` , defines a list of permission checks. The perspective will only be shown when at least one check fails.
+
+`required` and  `missing` can work together esp. in the case where you 
+want to show another perspective to users without specific permissions.
+
+If both `required` and `missing`
+are not specified for `state: AccessReview`, access review check will be ignored and perspective will be enabled.
```

Comment:
This seems unintuitive given the zero value of each check individually, see my comments on the API review for more

Reply:
As discussed with team, either one of required or missing will be present. Updated the doc as well.

Reply:
Regarding comment - https://github.com/openshift/enhancements/pull/1205#discussion_r966226188

The webhook cannot verify if the admin is trying to disable all the available perspectives because our perspectives are being contributed by dynamic extensions in the console code. We can definitely do this via the form experience.